sdks/java/io/csv/src/main/java/org/apache/beam/sdk/io/csv/CsvIOParseHelpers.java

```diff
@@ -48,8 +50,38 @@ static List<Schema.Field> mapFieldPositions(CSVFormat format, Schema schema) {
    * Parse the given {@link String} cell of the CSV record based on the given field's {@link
    * Schema.FieldType}.
    */
-  // TODO(https://github.com/apache/beam/issues/31719): implement method.
   static Object parseCell(String cell, Schema.Field field) {
-    return "";
+    Schema.FieldType fieldType = field.getType();
+    try {
+      switch (fieldType.getTypeName()) {
+        case STRING:
+          return cell;
+        case INT16:
+          return Short.parseShort(cell);
+        case INT32:
+          return Integer.parseInt(cell);
+        case INT64:
+          return Long.parseLong(cell);
+        case BOOLEAN:
+          return Boolean.parseBoolean(cell);
+        case BYTE:
+          return Byte.parseByte(cell);
+        case DECIMAL:
+          return new BigDecimal(cell);
+        case DOUBLE:
+          return Double.parseDouble(cell);
+        case FLOAT:
+          return Float.parseFloat(cell);
+        case DATETIME:
+          return Instant.parse(cell);
+        default:
+          throw new IllegalArgumentException(
```

Comment:
Consider UnsupportedOperationException ? Otherwise it will be caught in outer try block and throw again, which is redundant

Reply:
I completely agree, just changed the method to utilize UnsupportedOperationException instead.